src/main/webapp/app/quiz/edit/drag-and-drop-question/edit-drag-and-drop-question.component.ts

```diff
@@ -839,5 +804,59 @@ export class EditDragAndDropQuestionComponent implements OnInit, OnChanges, Afte
      */
     togglePreview(): void {
         this.showPreview = !this.showPreview;
+        this.prepareForSave();
+    }
+
+    /**
+     * @function changesInMarkdown
+     * @desc Detect of text changes in the markdown editor
+     *      1. Notify the parent component to check the validity of the text
+     *      2. Parse the text in the editor to get the newest values
+     */
+    changesInMarkdown(): void {
+        this.questionUpdated.emit();
+        this.changeDetector.detectChanges();
+        this.prepareForSave();
+    }
+
+    /**
+     * @function domainCommandsFound
+     * @desc 1. Gets the {array} containing the textLine with the domainCommandIdentifier and creates a new drag and drop problem statement
+     *       by assigning the textLines according to the domainCommandIdentifiers to the drag and drop attributes.
+     *       (question text, explanation, hint)
+     * @param {array} containing markdownTextLine with the corresponding domainCommand {DomainCommand} identifier
+     */
+    domainCommandsFound(domainCommands: [string, DomainCommand][]): void {
+        this.cleanupQuestion();
+        for (const  [textLine, command]  of domainCommands)  {
```

Comment:
as far as I understand `textLine` can also consist of multiple lines, so please refactor it into `text` instead